factory-reset-external

```diff
@@ -0,0 +1,124 @@
+#!/bin/sh
+#
+# External storage factory reset for Sailfish OS.
+#
+# Copyright (C) 2017 Jolla Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; version 2
+# of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+# Parameters:
+# $@ -> Device paths of partitions to reformat.
+
+format_script()
+{
+cat << 'EOF'
+#!/bin/sh
+
+function log
+{
+        echo "$0: $1"
+}
+
+for device in $@; do
+	unset TYPE
+	unset LABEL
+	unset UUID
+
+	if [ ! -e "$device" ]; then
+		continue
+	fi
+
+        eval "$(/sbin/blkid -c /dev/null -o export $device)"
+
+	case "$TYPE" in
+		vfat)
+			format_command=/sbin/mkfs.vfat
```

Comment:
do we have mkfs.vfat in initrd?

Reply:
In initrd, I think so actually.  But not in the rootfs where we're looking for it here.

Reply:
I take that back, initrd only has mkfs.ext4.